### ISSUE TITLE
Error in match arguments

### ISSUE THREAD

**Issue by susheelbhanu** [satijalab/azimuth#111]
Hi, 

I'm trying to annotate a seurate object and getting the following error. 

```
azimuth_gm.combined <- RunAzimuth(gm.combined, reference = "mousecortexref")
Error in match.arg(arg = tolower(x = type), choices = c("raw", "default",  : 
  'arg' should be one of “raw”, “default”, “NA”, “NA”
```

Could you please let me know how to fix this?

Thank you!

**Comment by AustinHartman** [comment]
Could you please post your `sessionInfo()`?

**Comment by susheelbhanu** [comment]
Here you go... Sorry it's bloated, 'cos I'm working on a few things simultaneously ;)
```
sessionInfo()
R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: OS X  12.4

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets  methods  
[9] base     

other attached packages:
 [1] biomaRt_2.46.3                  scmap_1.12.0                   
 [3] CHETAH_1.6.0                    SingleCellExperiment_1.12.0    
 [5] SummarizedExperiment_1.20.0     Biobase_2.50.0                 
 [7] GenomicRanges_1.42.0            GenomeInfoDb_1.26.7            
 [9] IRanges_2.24.1                  S4Vectors_0.28.1               
[11] BiocGenerics_0.36.0             MatrixGenerics_1.2.0           
[13] matrixStats_0.57.0              mousecortexref.SeuratData_1.0.0
[15] Azimuth_0.4.5                   shinyBS_0.61.1                 
[17] metap_1.7                       ifnb.SeuratData_3.1.0          
[19] SeuratData_0.2.1                fgsea_1.16.0                   
[21] msigdbr_7.5.1                   presto_1.0.0                   
[23] data.table_1.13.6               cowplot_1.1.1                  
[25] babelgene_22.3                  patchwork_1.1.1                
[27] sp_1.4-5                        SeuratObject_4.1.0             
[29] Seurat_4.1.1                    Rcpp_1.0.7                     
[31] reshape_0.8.8                   rstatix_0.7.0                  
[33] pairwiseAdonis_0.4              cluster_2.1.0                  
[35] vegan_2.5-7                     lattice_0.20-41                
[37] permute_0.9-5                   ggpubr_0.4.0                   
[39] RColorBrewer_1.1-2              knitr_1.39                     
[41] microbiome_1.12.0               phyloseq_1.34.0                
[43] ggrepel_0.9.1                   keyplayer_1.0.3                
[45] reshape2_1.4.4                  influenceR_0.1.0.1             
[47] ggsci_2.9                       Matrix_1.4-0                   
[49] SpiecEasi_1.1.0                 janitor_2.1.0                  
[51] forcats_0.5.1                   stringr_1.4.0                  
[53] dplyr_1.0.5                     purrr_0.3.4                    
[55] readr_1.4.0                     tidyr_1.1.3                    
[57] tibble_3.1.0                    ggplot2_3.3.3                  
[59] tidyverse_1.3.1                 igraph_1.2.6                   

loaded via a namespace (and not attached):
  [1] ica_1.0-2              class_7.3-17           ps_1.5.0              
  [4] foreach_1.5.1          lmtest_0.9-39          rprojroot_2.0.2       
  [7] glmnet_4.0-2           crayon_1.4.1           spatstat.core_2.3-2   
 [10] rbibutils_2.2.7        MASS_7.3-53            rhdf5filters_1.2.0    
 [13] nlme_3.1-151           backports_1.2.1        reprex_2.0.0          
 [16] huge_1.3.4.1           rlang_0.4.10           XVector_0.30.0        
 [19] ROCR_1.0-11            readxl_1.3.1           irlba_2.3.5           
 [22] callr_3.6.0            limma_3.46.0           BiocParallel_1.24.1   
 [25] bit64_4.0.5            glue_1.4.2             pheatmap_1.0.12       
 [28] sctransform_0.3.3      processx_3.5.1         AnnotationDbi_1.52.0  
 [31] spatstat.sparse_2.1-0  SeuratDisk_0.0.0.9020  shinydashboard_0.7.2  
 [34] spatstat.geom_2.4-0    VGAM_1.1-5             haven_2.3.1           
 [37] tidyselect_1.1.0       usethis_2.0.0          rio_0.5.16            
 [40] fitdistrplus_1.1-6     XML_3.99-0.5           zoo_1.8-8             
 [43] xtable_1.8-4           magrittr_2.0.1         Rdpack_2.1.3          
 [46] cli_2.4.0              zlibbioc_1.36.0        pulsar_0.3.7          
 [49] sn_2.0.1               rstudioapi_0.13        miniUI_0.1.1.1        
 [52] rpart_4.1-15           fastmatch_1.1-3        mathjaxr_1.4-0        
 [55] shiny_1.6.0            xfun_0.30              askpass_1.1           
 [58] pkgbuild_1.2.0         multtest_2.46.0        caTools_1.18.0        
 [61] biomformat_1.18.0      ape_5.4-1              listenv_0.8.0         
 [64] dendextend_1.14.0      Biostrings_2.58.0      png_0.1-7             
 [67] future_1.23.0          withr_2.3.0            bitops_1.0-6          
 [70] plyr_1.8.6             cellranger_1.1.0       e1071_1.7-7           
 [73] pillar_1.6.0           gplots_3.1.1           multcomp_1.4-15       
 [76] fs_1.5.0               hdf5r_1.3.5            googleVis_0.7.0       
 [79] vctrs_0.3.6            ellipsis_0.3.1         generics_0.1.0        
 [82] devtools_2.3.2         tools_4.0.2            foreign_0.8-81        
 [85] munsell_0.5.0          proxy_0.4-25           DelayedArray_0.16.0   
 [88] fastmap_1.1.0          compiler_4.0.2         pkgload_1.1.0         
 [91] abind_1.4-5            httpuv_1.5.5           sessioninfo_1.1.1     
 [94] plotly_4.9.3           rgeos_0.5-9            GenomeInfoDbData_1.2.4
 [97] gridExtra_2.3          deldir_1.0-6           mutoss_0.1-12         
[100] utf8_1.1.4             later_1.1.0.1          BiocFileCache_1.14.0  
[103] jsonlite_1.7.2         scales_1.1.1           pbapply_1.4-3         
[106] carData_3.0-4          lazyeval_0.2.2         promises_1.1.1        
[109] car_3.0-10             goftest_1.2-3          spatstat.utils_2.3-0  
[112] reticulate_1.22        openxlsx_4.2.3.9000    sandwich_3.0-0        
[115] Rtsne_0.15             uwot_0.1.11            survival_3.2-7        
[118] numDeriv_2016.8-1.1    plotrix_3.8-2          htmltools_0.5.1.1     
[121] memoise_1.1.0          viridisLite_0.3.0      digest_0.6.27         
[124] assertthat_0.2.1       mime_0.9               rappdirs_0.3.1        
[127] RSQLite_2.2.1          future.apply_1.8.1     remotes_2.2.0         
[130] blob_1.2.1             splines_4.0.2          labeling_0.4.2        
[133] Rhdf5lib_1.12.0        googledrive_1.0.1      RCurl_1.98-1.2        
[136] broom_0.7.6            hms_1.0.0              modelr_0.1.8          
[139] rhdf5_2.34.0           colorspace_2.0-0       BiocManager_1.30.10   
[142] mnormt_2.0.2           shape_1.4.5            tmvnsim_1.0-2         
[145] RANN_2.6.1             mvtnorm_1.1-1          fansi_0.4.1           
[148] parallelly_1.30.0      R6_2.5.0               grid_4.0.2            
[151] ggridges_0.5.2         lifecycle_1.0.0        zip_2.1.1             
[154] bioDist_1.62.0         TFisher_0.2.0          curl_4.3              
[157] ggsignif_0.6.3         googlesheets4_0.3.0    leiden_0.3.9          
[160] testthat_3.0.1         snakecase_0.11.0       desc_1.2.0            
[163] RcppAnnoy_0.0.19       TH.data_1.0-10         iterators_1.0.13      
[166] htmlwidgets_1.5.3      polyclip_1.10-0        rvest_1.0.0           
[169] mgcv_1.8-33            globals_0.14.0         openssl_1.4.3         
[172] progressr_0.10.0       codetools_0.2-18       lubridate_1.7.10      
[175] randomForest_4.6-14    gtools_3.8.2           prettyunits_1.1.1     
[178] dbplyr_2.1.1           gtable_0.3.0           DBI_1.1.0             
[181] tensor_1.5             httr_1.4.2             KernSmooth_2.23-18    
[184] stringi_1.5.3          progress_1.2.2         farver_2.0.3          
[187] viridis_0.5.1          DT_0.16                xml2_1.3.2            
[190] shinyjs_2.1.0          ade4_1.7-16            scattermore_0.7       
[193] bit_4.0.4              spatstat.data_2.1-2    pkgconfig_2.0.3       
[196] gargle_1.1.0           corrplot_0.84         
```

**Comment by AustinHartman** [comment]
Try installing v0.2.2 of SeuratData and let me know if that fixes the issue. I'll make sure to set the minimum version required by Azimuth in future updates

**Comment by susheelbhanu** [comment]
Okay, I have this: `SeuratData_0.2.2`, but still getting the same error.

**Comment by AustinHartman** [comment]
Thanks for trying that - I'm unable to replicate this error myself so I suspect it may have something to do with the Seurat object passed to  `RunAzimuth`. Any chance you could email me a small Seurat object I can reproduce the error with in order to debug? My email is ahartman@nygenome.org

**Comment by susheelbhanu** [comment]
Thank you. Sent you the data via email.

**Comment by AustinHartman** [comment]
Solved over email by upgrading SeuratData to v0.2.2 and restarting RStudio
